src/test/java/com/puppycrawl/tools/checkstyle/checks/header/HeaderCheckTest.java

```diff
@@ -114,7 +114,7 @@ public void testInvalidCharset() throws Exception {
                     + " com.puppycrawl.tools.checkstyle.checks.header.HeaderCheck"
                     + " - Cannot set property 'charset' to 'XSO-8859-1'",
                     ex.getMessage(), "Invalid exception message");
-            assertEquals("unsupported charset: 'XSO-8859-1'",
+            assertEquals("XSO-8859-1",
```

Comment:
@strkkk , please share full stack trace or explain why message become less clear.

Reply:
there will be a following message, since any exception is wrapped into CheckstyleException
```
com.puppycrawl.tools.checkstyle.api.CheckstyleException: cannot initialize module com.puppycrawl.tools.checkstyle.checks.header.HeaderCheck - Cannot set property 'charset' to 'XSO-8859-1'
```
so it is clear enough for user

Reply:
It is not clear enough as to why they can't set it. It will be OS/implementation specific, so what may work for one user, may not work for another so the new message does not give that full details as the old, imo.

Reply:
ok, I brought it back and changed UnsupportedEncodingException to UnsupportedCharsetException (which is used in forName and is unchecked) instead.